### PR TITLE
Fix server-side rendering of <select>

### LINKED ISSUE #1398: React.renderComponentToStaticMarkup doesn't generate selected attributes from value/defaultValue
I'm using static markup for unit testing [newforms](https://github/insin/newforms) output and I'd also like to support using it to render forms as static HTML.

Adding a `value` or `defaultValue` to a `<select>` doesn't generate static HTML with the appropriate `selected` attributes. However, they do get generated if you manually add them to `<option>` components, but then you get warning messages in the console when using the development build of React.

Is it possible to support both use cases without manually setting `selected` attributes, or will I just have to live with the warnings? :)

JSFiddle showing `<select>` static HTML output: http://jsfiddle.net/insin/2tvb7/3/

### PR DESCRIPTION
**Pull request by spicyj**
Fixes #1398.

### CODE REVIEW

File: src/browser/ui/dom/components/ReactDOMSelect.js [hunk 1]
```diff
@@ -133,19 +153,25 @@ var ReactDOMSelect = ReactClass.createClass({
 
   componentWillMount: function() {
     this._pendingUpdate = false;
-  },
 
-  componentDidMount: function() {
-    var value = LinkedValueUtils.getValue(this);
+    // Pass down initial value so initial generated markup has correct
+    // `selected` attributes
+    var value = LinkedValueUtils.getValue(this.props);
```
Comment (zpao) [inline src/browser/ui/dom/components/ReactDOMSelect.js]: Do you know why we didn't do this in getInitialState?

File: src/browser/ui/dom/components/ReactDOMSelect.js [hunk 2]
```diff
@@ -174,4 +200,6 @@ var ReactDOMSelect = ReactClass.createClass({
 
 });
 
+ReactDOMSelect.valueContextKey = valueContextKey;
```
Comment (zpao) [inline src/browser/ui/dom/components/ReactDOMSelect.js]: statics

File: src/browser/ui/dom/components/ReactDOMSelect.js (hunk 1 above)
Comment (spicyj) [inline src/browser/ui/dom/components/ReactDOMSelect.js]: No, I can move it.

**Comment by scottnonnenberg** [comment]
@spicyj Question: what implications, if any, does this have for server/client rendering? Is the generated code on the client different from what's generated on the server? Does this mean that the checksum will no longer match when React decides whether to re-render to the DOM page on the client?

**Comment by spicyj** [comment]
No, the same markup is still rendered on client and server.

**Comment by scottnonnenberg** [comment]
Awesome - thanks for the quick reply!

**Comment by zpao** [comment]
@spicyj - is this safe enough to take back in 0.13?

**Comment by spicyj** [comment]
Probably safer not to, but there's nothing in particular that ties this to 0.14.

**Comment by emptyemail** [comment]
Just curious if this is going to get released. Changing the order of items isn't really a great solution, and neither is living with console warnings. When prerendering forms, it causes a flicker when the react.render is finally called on the client and the form picks up the correct value.

**Comment by zpao** [comment]
It will be in 0.14.

**Pull request merged**